app/helpers/sessions_helper.rb

```diff
@@ -4,16 +4,71 @@ def log_in user
     session[:user_id] = user.id
   end
 
-  def current_user
-    @current_user ||= User.find_by id: session[:user_id]
-  end
-
   def logged_in?
     current_user.present?
   end
 
+  def forget user
+    user.forget
+    cookies.delete :user_id
+    cookies.delete :remember_token
+  end
+
   def log_out
-    reset_session
+    forget current_user
+    delete_session_and_cookies
     @current_user = nil
   end
+
+  def delete_session_and_cookies
+    session.delete(:user_id)
+    session.delete(:session_token)
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end
+
+  def current_user
+    @current_user ||= find_user_from_session || find_user_from_cookies
+  end
+
+  def remember user
+    user.remember
+    cookies.permanent.signed[:user_id] = user.id
+    cookies.permanent[:remember_token] = user&.remember_token
```

Comment:
```suggestion
    cookies.permanent[:remember_token] = user.remember_token
```
Dùng thừa, line phía trên k dùng toán tử an toàn, xuống line này lại dùng?